### PR TITLE
base16384: init at 2.2.2

### PR DESCRIPTION
**Pull request by Aleksanaa**
###### Description of changes

https://github.com/fumiama/base16384

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ocfox — APPROVED** [review]

File: pkgs/tools/text/base16384/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{ stdenv
+, lib
```
Comment (figsoda) [inline pkgs/tools/text/base16384/default.nix:2]:
```suggestion
{ lib
, stdenv
```

File: pkgs/tools/text/base16384/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+, fetchFromGitHub
+, cmake
+}:
```
Comment (figsoda) [inline pkgs/tools/text/base16384/default.nix:5]:
```suggestion
, installShellFiles
}:
```

File: pkgs/tools/text/base16384/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  version = "2.2.2";
+
+  src = fetchFromGitHub {
+    owner = "fumiama";
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    hash = "sha256-5Dfu88aAVtpNAwHAo+LOxe+jpgDhNdHYkWex42dCK0s=";
+  };
+
+  nativeBuildInputs = [ cmake ];
```
Comment (figsoda) [inline pkgs/tools/text/base16384/default.nix:18]:
```suggestion
  nativeBuildInputs = [ cmake installShellFiles ];

  postInstall = ''
    installManPage base16384.1
  '';
```

File: pkgs/tools/text/base16384/default.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+  meta = with lib; {
+    description = "Encode binary files to printable utf16be";
+    homepage = "https://github.com/fumiama/base16384";
+    changelog = "https://github.com/fumiama/base16384/releases/tag/v${version}";
+    license = licenses.gpl3Only;
```
Comment (figsoda) [inline pkgs/tools/text/base16384/default.nix:24]:
```suggestion
    license = licenses.gpl3Plus;
```

https://github.com/fumiama/base16384/commit/e270980b37cc23d6f69dc2a4c4feb0f676334369

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -1423,6 +1423,8 @@ with pkgs;
 
   asleap = callPackage ../tools/networking/asleap { };
 
+  base16384 = callPackage ../tools/text/base16384 { };
```
Comment (figsoda) [inline pkgs/top-level/all-packages.nix:1426]: there is a merge conflict

**Pull request closed without merging**